test/pkg.jl

```diff
@@ -139,6 +139,11 @@ temp_pkg_dir() do
     @test endswith(str, string(Pkg.installed("Example")))
     @test isempty(Pkg.dependents("Example"))
 
+    Pkg.checkout("Example", "test-branch") #17364
```

Comment:
did you just create this? if anyone deletes or commits to this branch would this test start failing?